modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DefaultDDMFormValuesFactory.java

```diff
@@ -42,16 +45,17 @@ public DDMFormValues create() {
 		ddmFormValues.setAvailableLocales(_ddmForm.getAvailableLocales());
 		ddmFormValues.setDefaultLocale(_ddmForm.getDefaultLocale());
 
-		for (DDMFormField ddmFormField : _ddmForm.getDDMFormFields()) {
-			DDMFormFieldValue ddmFormFieldValue =
-				createDefaultDDMFormFieldValue(ddmFormField);
-
-			ddmFormValues.addDDMFormFieldValue(ddmFormFieldValue);
-		}
+		populate(ddmFormValues);
 
 		return ddmFormValues;
 	}
 
+	public void populate(DDMFormValues ddmFormValues) {
```

Comment:
Hey, @carolmariaabb. Completely unrelated to the review and **not a priority at all**, I wanted to know your perspective on one decision I've made that I couldn't figure out a good enough reason to actually justify my first intention, _if it's not too much to ask of course_.

I kinda wanted to make this method `protected`, because, since this is a Factory class, it doesn't make much sense to have a `populate` method (at least in my opinion) alongside the `create`. Therefore, the `populate` method would be restricted to that package and this would be a good enough place to implement it, without "polluting" the code with one more class just to have this method hidden where no one would be able to find afterwards. What made me go with the `public` route is mainly because of Brian's approach to not having more of these `protected` methods and because the Factory is actually internal, so it's not a public API (only inside the "form-renderer" module). Do you have any thoughts on this? It's okay if you don't. I just wanted to get a peer's perspective on this. 😄

Thanks a lot for taking the time to read this! 😅

Reply:
Hi @rodrigopaulino 

> since this is a Factory class, it doesn't make much sense to have a populate method (at least in my opinion) alongside the create.

You made me think after this comment 😅. But I don't think it's wrong, considering the logic, because the `populate` is part of the process of creating the `DefaultDDMFormValues`.

> What made me go with the public route is mainly because of Brian's approach to not having more of these protected methods 

I was not aware of this. 🧐

>  because the Factory is actually internal, so it's not a public API 
I totally agree with you on this point. That's why I didn't see any problem with the method being public.

When I was reviewing the code, I thought about the possibility of leaving everything in the `create` method. 
In this case, we would need to do something like:
```
public DDMFormValues create() {
	if ((_ddmFormValues == null) ||
		ListUtil.isEmpty(_ddmFormValues.getDDMFormFieldValues())) {

		_ddmFormValues = new DDMFormValues(_ddmForm);

		_ddmFormValues.setAvailableLocales(_ddmForm.getAvailableLocales());
		_ddmFormValues.setDefaultLocale(_ddmForm.getDefaultLocale());
	}

	_populate(
		_ddmFormValues::addDDMFormFieldValue, _ddmForm.getDDMFormFields(),
		_ddmFormValues.getDDMFormFieldValuesMap(false));

	return _ddmFormValues;
}

public void setDDMFormValues(DDMFormValues ddmFormValues) {
	_ddmFormValues = ddmFormValues;
}
```
But I wasn't sure if this would be a better option than creating the new method `populate`.

Reply:
I've thought about putting the logic inside the `create` method as well, but I figured we would have to pass a `DDMFormValues` object into the Factory and it doesn't make much sense to me to have a `create` method that doesn't always "create". 😅

Reply:
That's exactly what I thought kkkk